contracts/potlock/src/potlock_interactions.rs

```diff
@@ -0,0 +1,106 @@
+use crate::potlock_requirements;
+use crate::potlock_storage::{self, Pot, Project};
+use crate::potlock_storage::{PotlockId, ProjectId};
+
+multiversx_sc::imports!();
+multiversx_sc::derive_imports!();
+
+#[multiversx_sc::module]
+pub trait PotlockInteractions:
+    potlock_requirements::PotlockRequirements
+    + potlock_storage::PotlockStorage
+    + multiversx_sc_modules::only_admin::OnlyAdminModule
+{
+    #[payable("*")]
+    #[endpoint(addPot)]
+    fn add_pot(&self, name: ManagedBuffer, description: ManagedBuffer) {
+        let payment_for_adding_pot = self.call_value().single_esdt();
+        require!(
+            self.fee_token_identifier().get() == payment_for_adding_pot.token_identifier,
+            "Wrong token identifier for creating a pot!"
+        );
+        require!(
+            self.fee_amount().get() == payment_for_adding_pot.amount,
+            "Wrong fee amount for creating a pot"
+        );
+        let caller = self.blockchain().get_caller();
+
+        let potlock_id = self.potlocks().len() + 1;
+        let potlock = Pot::new(potlock_id, caller, name, description);
+        self.potlocks().push(&potlock);
+    }
+
+    #[endpoint(applyForPot)]
+    fn apply_for_pot(
+        &self,
+        potlock_id: PotlockId,
+        project_name: ManagedBuffer,
+        description: ManagedBuffer,
+    ) -> usize {
+        let owner = self.blockchain().get_caller();
+        let project = Project::new(potlock_id, project_name, description, owner);
+        self.projects().push(&project)
+    }
+
+    #[payable("*")]
+    #[endpoint(donateToPot)]
+    fn donate_to_pot(&self, potlock_id: PotlockId) {
+        self.require_potlock_exists(potlock_id);
+        self.require_potlock_is_active(potlock_id);
+
+        let payment = self.call_value().single_esdt();
```

Comment:
Do you need to accept NFT payments? If not, I would change this to single_fungible_esdt.